prov/util/src/util_shm.c

```diff
@@ -419,6 +425,11 @@ int smr_map_to_region(const struct fi_provider *prov, struct smr_peer *peer_buf)
 	peer = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0);
 	peer_buf->region = peer;
 
+	ret = ofi_hmem_host_register(peer, peer->total_size);
```

Comment:
Is it possible to only do this when HMEM is enabled?

Reply:
It would require some extra parameters passed around but I can add that. Good call

Reply:
@j-xiong Had to add some flags to pass this around. Not the prettiest thing but it's there.